tms_django/apps/tms/models.py

```diff
@@ -1,3 +1,115 @@
-from django.db import models
+from __future__ import annotations
 
-# Create your models here.
+from functools import partial
+
+from django.apps import apps
+from django.conf import settings
+from django.core.exceptions import ObjectDoesNotExist
+from django.core.validators import MinValueValidator
+from django.db import OperationalError, models
+from django.utils.translation import gettext_lazy as _
+from loguru import logger
+
+
+def generate_model_custom_id(model_name: str, custom_id_field_name: str, default_value: int) -> str:
+    model = apps.get_model(app_label="tms", model_name=model_name)
+
+    try:
+        instance = model.objects.order_by("-id")[:1].get()
+    except (OperationalError, ObjectDoesNotExist):
+        logger.info(f"No record found for {model_name}, setting {default_value=}")
+        last_custom_id = default_value
+
+    else:
+        last_custom_id = getattr(instance, custom_id_field_name)
+        logger.info(f"Last id for {model_name}: {last_custom_id}")
+        last_custom_id = int(last_custom_id) + 1
+
+    return str(last_custom_id)
+
+
+class TruckKind(models.TextChoices):
+    SPRINTER = "sprinter"
+    SMALL_STRAIGHT = "small_straight"
+    LARGE_STRAIGHT = "large_straight"
+
+
+class Truck(models.Model):
+    kind = models.CharField(choices=TruckKind, max_length=255)
+    weight = models.IntegerField(null=True, default=None, validators=[MinValueValidator(0)])
+    capacity = models.IntegerField(null=True, default=None, validators=[MinValueValidator(0)])
+
+    def __str__(self) -> str:
+        return f"{self.kind}"
+
+
+generate_unit_id = partial(
+    generate_model_custom_id, model_name="DriverProfile", custom_id_field_name="unit_id", default_value=1000
+)
+
+
+class DriverProfile(models.Model):
+    user = models.OneToOneField(settings.AUTH_USER_MODEL, related_name="driver_profile", on_delete=models.CASCADE)
+    unit_id = models.CharField(max_length=6, unique=True, db_index=True, default=generate_unit_id)
+    phone_number = models.CharField(max_length=255, blank=False)
+    driver_licence = models.CharField(max_length=255, blank=False)
+    date_of_birth = models.DateField(null=True, blank=False)
+    address = models.CharField(max_length=255)
+    truck = models.ForeignKey(Truck, null=True, on_delete=models.CASCADE)
```

Comment:
@elya-hp Does it mean that a driver may have a link to only one truck? Is linking with 2 or more trucks not viable?

Reply:
Correct. He can be assigned only 1 truck